src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -8156,16 +8154,20 @@ private static string WinInternalGetLinkType(string filePath)
                     // Get Buffer size
                     IntPtr dangerousHandle = handle.DangerousGetHandle();
 
-                    bool result = DeviceIoControl(dangerousHandle, FSCTL_GET_REPARSE_POINT,
-                        IntPtr.Zero, 0, outBuffer, outBufferSize, out bytesReturned, IntPtr.Zero);
+                    bool result = DeviceIoControl(
+                        dangerousHandle,
+                        FSCTL_GET_REPARSE_POINT,
+                        InBuffer: IntPtr.Zero,
+                        nInBufferSize: 0,
+                        outBuffer,
+                        outBufferSize,
+                        out bytesReturned,
+                        lpOverlapped: IntPtr.Zero);
 
                     if (!result)
```

Comment:
It seems that in the future, if new reparse types (or 3rd party types) are added, then the struct may change so it's possible to get an error indicating the buffer size is too small and it would just get swallowed here.  My preference would be to check for specific errors like ERROR_NOT_A_REPARSE_POINT and ERROR_INVALID_FUNCTION and then return null, otherwise throw the exception so that we can update the code.

Reply:
Here we need only _positive_ result "Yes, it is a reparse point". Any change of the behavior on OS level will be a breaking change.